### PR TITLE
Refactoring dem

### PR DESCRIPTION
**Pull request by shahabgol**
Move some functions into separate files to decrease the length of dem class

### CODE REVIEW

File: include/dem/checkpointing.h [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+/**
+ * Carries out reading and writing dem checkpoints
+ *
+ * @note
+ *
+ * @author Shahab Golshan, Polytechnique Montreal 2019-
+ */
+
+template <int dim>
+class Checkpointing
```
Comment (blaisb) [inline include/dem/checkpointing.h:48]: Clean stuff. Maybe class should be called DEMCheckpointing?

File: include/dem/print_initial_information.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * Author: Shahab Golshan, Polytechnique Montreal, 2019
+ */
+
+#include <dem/dem_solver_parameters.h>
+
+using namespace std;
+
+#ifndef print_initial_information_h
+#  define print_initial_information_h
```
Comment (blaisb) [inline include/dem/print_initial_information.h:25]:
```suggestion
#define print_initial_information_h
```

File: include/dem/checkpointing.h (hunk 1 above)
Comment (blaisb) [reply include/dem/checkpointing.h:48]: Actually, why does this need to be a class? Could it not be just two functions? Class logic does not appear to be used here.

File: source/dem/read_mesh.cc [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#include <dem/read_mesh.h>
+
+template <int dim>
+void
+read_mesh(const DEMSolverParameters<dim> &           parameters,
+          const ConditionalOStream &                 pcout,
+          parallel::distributed::Triangulation<dim> &triangulation,
+          double &triangulation_cell_diameter)
```
Comment (blaisb) [inline source/dem/read_mesh.cc:8]: How is this function different than the read mesh function that is used in the core lethe library?

**Review by blaisb — COMMENTED** [review]
Some minor comments:
- Checkpointing in it's present state does not need to be a class
- Can you see what is the difference between your read dem mesh and the one in the core library? If they are similar or if the core can be adaptated instead, we should use the one in the core. If there is a difference that make it not possible, then we can keep two but this will be a bit confusing...

File: include/dem/checkpointing.h (hunk 1 above)
Comment (shahabgol) [reply include/dem/checkpointing.h]: done

**Review by blaisb — APPROVED** [review]

**Pull request merged**